### PR TITLE
FACT-652 adding bespoke tabs component

### PR DESCRIPTION
**Pull request by Majid-justice**
https://tools.hmcts.net/jira/browse/FACT-652

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/assets/js/fact-tabs/fact-tabs.js [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    // Remove events
+    $tab.removeEventListener('click', $tab.boundTabClick, true);
+    $tab.removeEventListener('keydown', $tab.boundTabKeydown, true);
+
+    // Unset HTML attributes
+    this.unsetAttributes($tab);
+  }.bind(this));
+
+  // Remove hashchange event handler
+  window.removeEventListener('hashchange', $module.boundOnHashChange, true);
```
Comment (agraham01) [inline src/main/assets/js/fact-tabs/fact-tabs.js:110]: The current GDS component appends a hash on to the URL and when browser history or the forward/back buttons are used, it navigates back to the appropriate tab. The hash is present in the new implementation but the forward/back button navigation is broken.

File: src/main/assets/js/fact-tabs/fact-tabs.js [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  return hash;
+};
+
+FactTabs.prototype.getSelectedTab = function() {
+  var text = this.$module.querySelector('.fact-tabs-title').innerText;
+  return this.$module.querySelector('.fact-tabs-tab').innerText = text;
+};
+
+FactTabs.prototype.setSelected = function(string){
+  var selectedTab = this.$module.querySelector('.fact-tabs-title');
```
Comment (agraham01) [inline src/main/assets/js/fact-tabs/fact-tabs.js:273]: When you select the tab in the list, the list/dropdown doesn't disappear; you have to move the mouse away. I wonder if it would be more user-friendly for the list/dropdown to close.

File: src/main/assets/js/fact-tabs/fact-tabs.js [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+  selectedTab.style.paddingTop = '20px';
+  selectedTab.style.paddingBottom= '11px';
```
Comment (agraham01) [inline src/main/assets/js/fact-tabs/fact-tabs.js:275]: Should we move this to the css?

File: src/main/assets/scss/tabs.scss [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+      padding: govuk-spacing(2) govuk-spacing(4);
+      position: relative;
+      text-align: left;
+
+      &::before {
+        content: none;
+      }
+
+      &:hover {
+        background-color: $blue;
```
Comment (agraham01) [inline src/main/assets/scss/tabs.scss:133]: The blue background over black text doesn't provide much contrast.

File: src/test/functional/steps/local-authorities.ts [hunk 5]
```diff
@@ -65,24 +77,29 @@ Then('An error is displayed for local authorities with title {string} and summer
...
-  const elementFamilyCourtChecked = await I.isElementChecked(selectorFamilyCourt);
-  if (elementFamilyCourtChecked) {
-    await I.click(selectorFamilyCourt);
-  }
-  const elementTribunalCourtChecked = await I.isElementChecked(selectorTribunalCourt);
-  if (!elementTribunalCourtChecked) {
-    await I.click(selectorTribunalCourt);
+  const elementVisible = await I.isElementVisible(selectorFamilyCourt);
+
+  if (elementVisible) {
```
Comment (agraham01) [inline src/test/functional/steps/local-authorities.ts:82]: I'm not sure why this change has been made - could you explain? If the element isn't visible, the 'checkElement' call won't be made so the step won't fail.

Comment (Majid-justice) [reply src/test/functional/steps/local-authorities.ts:82]: I was getting a issue that the check for the element was happen before the page fully loaded so the test kept falling. So I put in a extra check for visibility which makes sure the element  is loaded fully

Comment (agraham01) [reply src/test/functional/steps/local-authorities.ts:82]: I tried running the test in debug mode and the elementVisible was 'false' so the code in the block wasn't run.

Comment (Majid-justice) [reply src/test/functional/steps/local-authorities.ts]: I have removed the elementVisible variable and fixed the issue

File: src/main/assets/scss/tabs.scss (hunk 4 above)
Comment (Majid-justice) [reply src/main/assets/scss/tabs.scss:134]: I have changed the text coloraturas to white on hover for better contrast

File: src/main/assets/js/fact-tabs/fact-tabs.js (hunk 3 above)
Comment (Majid-justice) [reply src/main/assets/js/fact-tabs/fact-tabs.js:275]: needs this to be there as issue with padding being removed when inner text property updated

File: src/main/assets/js/fact-tabs/fact-tabs.js (hunk 2 above)
Comment (Majid-justice) [reply src/main/assets/js/fact-tabs/fact-tabs.js:273]: would need to change it from hover on to a click event to achieve that

File: src/main/assets/js/fact-tabs/fact-tabs.js (hunk 1 above)
Comment (Majid-justice) [reply src/main/assets/js/fact-tabs/fact-tabs.js:110]: The up and down arrows seem to working on my end tried on chrome and safari

Comment (agraham01) [reply src/main/assets/js/fact-tabs/fact-tabs.js:110]: I'm referring to browser navigation e.g. using history, bookmarked URL or the browser's forward and back buttons. It doesn't seem to be working in Chrome or Safari for me.

Comment (Majid-justice) [reply src/main/assets/js/fact-tabs/fact-tabs.js:113]: I have added forward/back navigation for browser by adding missing event handler

**Review by agraham01 — APPROVED** [review]
